packages/calimero-sdk/src/wallets/StarknetLogin/LoginWithStarknet.tsx

```diff
@@ -0,0 +1,390 @@
+    try {
+      setErrorMessage(null);
+      setLoading(true);
+      if(starknetAccount) {
+        const message = {
+          domain: {
+            name: "ServerChallenge",
+            chainId: "SN_MAIN",
+            version: "1",
+            revision: "1"
+          },
+          types: {
+            StarknetDomain: [
+              { name: "name", type: "shortstring" },
+              { name: "chainId", type: "shortstring" },
+              { name: "version", type: "shortstring" },
+              { name: "revision", type: "shortstring" },
+            ],
+            Challenge: [
+              { name: "nodeSignature", type: "string"},
+              { name: "publicKey", type: "string"},
+            ],
+          },
+          primaryType: "Challenge",
+          message: {
+            nodeSignature: walletSignatureData.payload.message.nodeSignature,
+            publicKey: walletSignatureData.payload.message.publicKey
+          }
+        };
+        const signature = await starknetAccount.account.signMessage(message);
+        const messageHash = await starknetAccount.account.hashMessage(message);
+        if (signature) {
+          setSignData({
+            signature: signature,
+            messageHash: messageHash,
+          });
+        }
+      }
+    }catch(error) {
+      console.error('Error signing message:', error);
+      setErrorMessage('Error signing message');
+    }
+    setLoading(false);
+  }, [starknetAccount, walletSignatureData]);
+
+  const login = useCallback(async () => {
+    try {
+      setErrorMessage(null);
+      setLoading(true);
+      if (!signData) {
+        console.error('signature is empty');
+        setErrorMessage('Signature is empty');
+      } else if (!starknetAccount) {
+        console.error('address is empty');
+        setErrorMessage('Address is empty');
+      } else {
+        const walletMetadata: WalletMetadata = {
+          wallet: getWalletType(starknetAccount?.id),
+          signingKey: starknetAccount?.id === 'argentX' ? starknetAccount?.account.address : await starknetAccount?.account.signer.getPubKey(),
+          walletAddress: starknetAccount.account.address
+        };
+        const loginRequest: LoginRequest = {
+          walletSignature: signData,
+          payload: walletSignatureData?.payload,
+          walletMetadata: walletMetadata,
+          contextId,
+        };
+        await apiClient
+          .node()
+          .login(loginRequest, rpcBaseUrl)
+          .then((result) => {
+            if (result.error) {
+              console.error('Login error: ', result.error);
+              setErrorMessage(result.error.message);
+            } else {
+              setStorageNodeAuthorized();
+              successRedirect();
+            }
+          })
+          .catch(() => {
+            console.error('error while login!');
+            setErrorMessage('Error while login!');
+          });
+      }
+    }catch(error) {
```

Comment:
```suggestion
    } catch(error) {
```
Use lint to format code